### ISSUE TITLE
Youtube link has either no audio or no image

### ISSUE THREAD

**Issue by xundinha** [streamlink/streamlink#2284]
If i use streamlink (youtubelink) best  it has no audio

If i use streamlink (youtubelink) audio_mp4 it has no image 
and i couldnt figure a way to use both commands. Is there a command to fix this?

Windows 64
Using the 0.14.2 version 

it just doesnt work if i use 1080p and 720p60
do have ffmpeg installed
Thanks

**Comment by beardypig** [comment]
You should follow the template as it really helps us identify possible issues.

Do you have `ffmpeg` installed? Are you using Windows, Linux or macOS, which version of streamlink do you have. You *must* provide all the information requested in the template. 

If you don't read the template you won't get much help.

**Comment by xundinha** [comment]
well i kinda fixed it, if i dont use 1080p it plays the audio

**Comment by xundinha** [comment]
oh i'm using windows sorry, i have the latest version just download yesterday sry about that

**Comment by beardypig** [comment]
Can you provide a debug log. `streamlink -l debug youtube.com/... best`

**Comment by xundinha** [comment]
C:\Users\Chatubao>streamlink -l debug https://www.youtube.com/watch?v=CQPCL2HIuhY best
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.6.6
[cli][debug] Streamlink: 1.0.0
[cli][debug] Requests(2.21.0), Socks(1.6.7), Websocket(0.54.0)
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=CQPCL2HIuhY
[plugin.youtube][debug] Video ID from URL
[plugin.youtube][debug] Using video ID: CQPCL2HIuhY
[plugin.youtube][debug] get_video_info - 1: Found data
[plugin.youtube][debug] MuxedStream: v 302 a 251 = 720p60
[cli][info] Available streams: audio_mp4, audio_webm, 144p (worst), 180p, 360p, 720p, 720p60 (best)
[cli][info] Opening stream: 720p60 (muxed-stream)
[stream.ffmpegmux][debug] Opening http substream
[stream.ffmpegmux][debug] Opening http substream
[stream.mp4mux-ffmpeg][debug] ffmpeg command: C:\Program Files (x86)\Streamlink\ffmpeg\ffmpeg.exe -nostats -y -i \\.\pipe\ffmpeg-5816-816 -i \\.\pipe\ffmpeg-5816-991 -c:v copy -c:a copy -map 0 -map 1 -f matroska pipe:1
[stream.ffmpegmux][debug] Starting copy to pipe: \\.\pipe\ffmpeg-5816-816
[stream.ffmpegmux][debug] Starting copy to pipe: \\.\pipe\ffmpeg-5816-991
[cli][debug] Pre-buffering 8192 bytes
[cli][info] Starting player: "C:\Program Files\VideoLAN\VLC\vlc.exe"
[cli.output][debug] Opening subprocess: "C:\Program Files\VideoLAN\VLC\vlc.exe" --input-title-format https://www.youtube.com/watch?v=CQPCL2HIuhY -
[cli][debug] Writing stream to output
[stream.ffmpegmux][debug] Closing ffmpeg thread
[stream.ffmpegmux][debug] Pipe copy complete: \\.\pipe\ffmpeg-5816-816
Interrupted! Exiting...
[cli][info] Closing currently open stream...
[stream.ffmpegmux][debug] Closing ffmpeg thread

**Comment by beardypig** [comment]
OK. Which version of VLC are you using? If you are using VLC3 try VLC2, if you are using VLC2 try VLC3 :)

**Comment by xundinha** [comment]
ok thank you gonna do that

**Comment by xundinha** [comment]
weird if i use 2.0 if doesnt show no image and no audio.
just gonna watch stuff on 720p thanks !

**Comment by beardypig** [comment]
You might need to use a different version of VLC3, there have been a few issues with YouTube streams when using VLC.

**Comment by gravyboat** [comment]
I'm going to close this as the question has been answered.

**Comment by Victor333Huesca** [comment]
The issue seems to be present in version 1.1.1.

### Description
There is no audio for _muxed stream_ (auto-selected for 1080p youtube videos).
The audio is present for lower resolution which use _http_.
The problem seems to be related to the _stream_ in use, not the youtube extension.
I think _ffmpegmux_ is the key but I don't understand ffmpeg enough to know what is wrong.

### Informations
- Plateform: Windows 10
- VLC: 3.06
- Streamlink: 1.1.1

### Log
On _muxed-stream_ (1080p)
```
C:\Users\Victor>streamlink https://www.youtube.com/watch?v=SfLVlT8SkrE 1080p --loglevel debug
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.6.6
[cli][debug] Streamlink: 1.1.1
[cli][debug] Requests(2.21.0), Socks(1.6.7), Websocket(0.56.0)
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=SfLVlT8SkrE
[plugin.youtube][debug] Video ID from URL
[plugin.youtube][debug] Using video ID: SfLVlT8SkrE
[plugin.youtube][debug] get_video_info - 1: Found data
[plugin.youtube][debug] MuxedStream: v 137 a 251 = 1080p
[cli][info] Available streams: audio_mp4, audio_webm, 360p (worst), 720p, 1080p (best)
[cli][info] Opening stream: 1080p (muxed-stream)
[stream.ffmpegmux][debug] Opening http substream
[stream.ffmpegmux][debug] Opening http substream
[stream.mp4mux-ffmpeg][debug] ffmpeg command: C:\Program Files (x86)\Streamlink\ffmpeg\ffmpeg.exe -nostats -y -i \\.\pipe\ffmpeg-15332-559 -i \\.\pipe\ffmpeg-15332-843 -c:v copy -c:a copy -map 0 -map 1 -f matroska pipe:1
[stream.ffmpegmux][debug] Starting copy to pipe: \\.\pipe\ffmpeg-15332-559
[stream.ffmpegmux][debug] Starting copy to pipe: \\.\pipe\ffmpeg-15332-843
[cli][debug] Pre-buffering 8192 bytes
[cli][info] Starting player: "C:\Program Files\VideoLAN\VLC\vlc.exe"
[cli.output][debug] Opening subprocess: "C:\Program Files\VideoLAN\VLC\vlc.exe" --input-title-format https://www.youtube.com/watch?v=SfLVlT8SkrE -
[cli][debug] Writing stream to output
```

On _http_ (720p)
```
C:\Users\Victor>streamlink https://www.youtube.com/watch?v=SfLVlT8SkrE 720p -l debug
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.6.6
[cli][debug] Streamlink: 1.1.1
[cli][debug] Requests(2.21.0), Socks(1.6.7), Websocket(0.56.0)
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=SfLVlT8SkrE
[plugin.youtube][debug] Video ID from URL
[plugin.youtube][debug] Using video ID: SfLVlT8SkrE
[plugin.youtube][debug] get_video_info - 1: Found data
[plugin.youtube][debug] MuxedStream: v 137 a 251 = 1080p
[cli][info] Available streams: audio_mp4, audio_webm, 360p (worst), 720p, 1080p (best)
[cli][info] Opening stream: 720p (http)
[cli][debug] Pre-buffering 8192 bytes
[cli][info] Starting player: "C:\Program Files\VideoLAN\VLC\vlc.exe"
[cli.output][debug] Opening subprocess: "C:\Program Files\VideoLAN\VLC\vlc.exe" --input-title-format https://www.youtube.com/watch?v=SfLVlT8SkrE -
[cli][debug] Writing stream to output
```

**Comment by bastimeyer** [comment]
@Victor333Huesca 
The muxed stream does have an audio channel. It's working fine in MPV...

```
$ streamlink --stdout 'https://www.youtube.com/watch?v=SfLVlT8SkrE' 1080p | ffprobe -v error -show_format -show_streams -print_format json - | jq
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=SfLVlT8SkrE
[cli][info] Available streams: audio_mp4, audio_webm, 360p (worst), 720p, 1080p (best)
[cli][info] Opening stream: 1080p (muxed-stream)
{
  "streams": [
    {
      "index": 0,
      "codec_name": "h264",
      "codec_long_name": "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
      "profile": "High",
      "codec_type": "video",
      "codec_time_base": "1/50",
      "codec_tag_string": "[0][0][0][0]",
      "codec_tag": "0x0000",
      "width": 1920,
      "height": 1080,
      "coded_width": 1920,
      "coded_height": 1088,
      "has_b_frames": 1,
      "sample_aspect_ratio": "1:1",
      "display_aspect_ratio": "16:9",
      "pix_fmt": "yuv420p",
      "level": 40,
      "color_range": "tv",
      "color_space": "bt709",
      "color_transfer": "bt709",
      "color_primaries": "bt709",
      "chroma_location": "left",
      "field_order": "progressive",
      "refs": 1,
      "is_avc": "true",
      "nal_length_size": "4",
      "r_frame_rate": "25/1",
      "avg_frame_rate": "25/1",
      "time_base": "1/1000",
      "start_pts": 0,
      "start_time": "0.000000",
      "bits_per_raw_sample": "8",
      "disposition": {
        "default": 1,
        "dub": 0,
        "original": 0,
        "comment": 0,
        "lyrics": 0,
        "karaoke": 0,
        "forced": 0,
        "hearing_impaired": 0,
        "visual_impaired": 0,
        "clean_effects": 0,
        "attached_pic": 0,
        "timed_thumbnails": 0
      },
      "tags": {
        "HANDLER_NAME": "ISO Media file produced by Google Inc. Created on: 05/02/2019."
      }
    },
    {
      "index": 1,
      "codec_name": "opus",
      "codec_long_name": "Opus (Opus Interactive Audio Codec)",
      "codec_type": "audio",
      "codec_time_base": "1/48000",
      "codec_tag_string": "[0][0][0][0]",
      "codec_tag": "0x0000",
      "sample_fmt": "fltp",
      "sample_rate": "48000",
      "channels": 2,
      "channel_layout": "stereo",
      "bits_per_sample": 0,
      "r_frame_rate": "0/0",
      "avg_frame_rate": "0/0",
      "time_base": "1/1000",
      "start_pts": -7,
      "start_time": "-0.007000",
      "disposition": {
        "default": 1,
        "dub": 0,
        "original": 0,
        "comment": 0,
        "lyrics": 0,
        "karaoke": 0,
        "forced": 0,
        "hearing_impaired": 0,
        "visual_impaired": 0,
        "clean_effects": 0,
        "attached_pic": 0,
        "timed_thumbnails": 0
      },
      "tags": {
        "language": "eng"
      }
    }
  ],
  "format": {
    "filename": "pipe:",
    "nb_streams": 2,
    "nb_programs": 0,
    "format_name": "matroska,webm",
    "format_long_name": "Matroska / WebM",
    "start_time": "-0.007000",
    "probe_score": 100,
    "tags": {
      "COMPATIBLE_BRANDS": "iso6avc1mp41",
      "MAJOR_BRAND": "dash",
      "MINOR_VERSION": "0",
      "ENCODER": "Lavf58.20.100"
    }
  }
}
error: Error when writing to output: [Errno 32] Broken pipe, exiting
[stream.ffmpegmux][error] Pipe copy aborted: /tmp/ffmpeg-22749-932
[stream.ffmpegmux][error] Pipe copy aborted: /tmp/ffmpeg-22749-126
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

**Comment by Victor333Huesca** [comment]
You are right, I tried with _mpv_ and the audio was working.
I think it is a vlc issue.
Thank you.

**Comment by pcgeek86** [comment]
I have the same problem on Windows 10, where the 1080p stream doesn't play audio correctly. In the VLC statistics window, I can see that there is an audio stream, using the Opus format, but none of the audio blocks are actually being played back.

* Windows 10 build 1909
* nVidia GeForce GTX 1080
* Streamlink 1.4.0
* VLC 3.0.8 Vetinari

I was able to work around this problem by using the Streamlink `--ffmpeg-audio-transcode` parameter. Here's an example of a working solution:

```
.\streamlink.exe --ffmpeg-audio-transcode mp3 https://www.youtube.com/watch?v=Gf-xHknIS9g best
```

I ran into a separate problem, where 2160p (4k UHD) streams are not available, unless they're 60 fps. 2160p60 works, but YouTube videos that are in 4k 30 fps (2160p) doesn't even show up in Streamlink. I'll have to open a new issue for this.
